packages/coverage-istanbul/src/provider.ts

```diff
@@ -248,17 +260,39 @@ export class IstanbulCoverageProvider extends BaseCoverageProvider implements Co
         })
       }
     }
+  }
 
-    // In watch mode we need to preserve the previous results if cleanOnRerun is disabled
-    const keepResults = !this.options.cleanOnRerun && this.ctx.config.watch
+  async mergeReports(path: string) {
+    const directory = resolve(path, 'coverage')
+    const files = await fs.readdir(directory)
+    const coverageMap = libCoverage.createCoverageMap({})
 
-    if (!keepResults) {
-      this.coverageFiles = new Map()
-      await fs.rm(this.coverageFilesDirectory, { recursive: true })
+    for (const file of files) {
+      const report = await fs.readFile(resolve(directory, file), 'utf8')
+      coverageMap.merge(JSON.parse(report))
+    }
+
+    await this.generateReports(coverageMap, true)
+  }
+
+  async createBlobReport(coverageMap: CoverageMap, options: Options['reporter'][number][1]) {
+    const dir = this.ctx.config.coverage.reportsDirectory || '.vitest-reports/coverage/'
```

Comment:
Using `config.coverage.reportsDirectory` with `blob` reporter is a bit tricky. By default coverage providers always clean up `reportsDirectory` before starting the tests, so earlier blob-reports are cleaned as well. Maybe it's better to use reporter options for this, e.g.:

```diff
- reporters: [['blob', { file: 'blob.json' }]],
+ reporters: [['blob', { outputFile: '.vitest-reports/coverage/blob.json' }]],
```

But there's no way to pass this via CLI at the moment.